src/js/hooks/cycleCount/useResolveStep.js

```diff
@@ -712,10 +718,13 @@ const useResolveStep = () => {
     // Nested path in colum names contains "_" instead of "."
     const nestedPath = columnId.replaceAll('_', '.');
     // Update data for: cycleCount (table) -> cycleCountItem (row) -> column (nestedPath)
+    const valueChanged = _.get(tableData.current, `[${tableIndex}].cycleCountItems[${rowIndex}].${nestedPath}`) !== value;
```

Comment:
there was another bug with the `console.log` not displaying the live data (ref: https://github.com/openboxes/openboxes/pull/5215#discussion_r2048905176)
it was difficult to determine what the issue is, but it turned out that `updateData` method is called a lot of times after each rerender, not only when we actually update the data, so it always set the `updated` flag to `true` even if something a while ago set it to `false` (`markAllItemsAsNotUpdated` method).
So this `valueChanged` flag is needed to determine if a value really changed and only then update the `updated` flag to `true`.